### ISSUE TITLE
DiffTest.py:189: python3: undefined symbol: clang_annotateTokens

### ISSUE THREAD

**Issue by jayvdb** [coala/coala#3182]
https://travis-ci.org/jayvdb/coala/builds/182528852 - Trusty build

```
Collecting libclang-py3~=0.2.0 (from -r requirements.txt (line 5))
  Downloading libclang-py3-0.2.tar.gz
...
SKIP [1] tests/results/DiffTest.py:189: python3: undefined symbol: clang_annotateTokens. Please ensure that your python bindings are compatible with your libclang.so version.
```

**Comment by jayvdb** [comment]
This is caused by the Travis workers having a different clang installed.
The first step is to use the correct python package version; ala https://github.com/coala/coala/issues/3187 .
Then I do not know how we support multiple versions of clang , except to remove the versioned requirement of libclang-py3 , allowing it to be whatever the user needs to install in order to match their clang version.

**Comment by vijeth-aradhya** [comment]
@jayvdb If travis is taking installation details from requirements.txt, then it should install `libclang-py3~=3.4.0` right? How is it installing `libclang-py3~=0.2.0` - is it because it has a different clang version?
In that case, we should allow the users to install the above package according to the clang version they have... I'd like to take up this issue :+1:

**Comment by jayvdb** [comment]
See the second comment.
> The first step is to use the correct python package version; ala #3187 .

that has been done now, which is why you see `libclang-py3~=3.4.0` in `requirements.txt`. (see https://github.com/coala/coala/commit/f25d48daf5a)

> In that case, we should allow the users to install the above package according to the clang version they have... I'd like to take up this issue

First we need to know what works.
Is it possible to use clang 3.6 and libclang-py3 3.6 ?  or 3.8 ? or 3.9?

**Comment by vijeth-aradhya** [comment]
@jayvdb [Here](https://bitbucket.org/Anteru/python3-libclang), it says libclang-py3 3.9 is compatible with Clang 3.9
My guess is that each release is compatible with the corresponding release of clang
We could test it out though. ( if clang 3.6 is compatible with 3.8 or 3.9 )
In any case we can handle the testcases on our builds in a particular clang and libclang-py3 version and leave the user to install the correct version right

**Comment by jayvdb** [comment]
> We could test it out though. 

Yes, that is what needs doing.  Nothing else is worth discussing until we know what works local on our workstations or on Travis (Ubuntu and OSX).

**Comment by damngamerz** [comment]
Im able to reproduce this issue in my local system.I was trying to solve it by using libclang-py3-3.6.0.But it throws another exception.With 27 Tests failed and majority of it has.
```
pkg_resources.ContextualVersionConflict: (libclang-py3 3.6.0 (/usr/local/lib/python3.6/site-packages), Requirement.parse('libclang-py3~=3.4.0'), {'coala'})
```
what does it mean any other way to solve this issue.

**Comment by jack17529** [comment]
@damngamerz  I am still getting this error ,are you working on this ?

**Comment by damngamerz** [comment]
@jack17529 go ahead if you have a solution :smile: you should assign yourself first before submitting a PR.

**Comment by jack17529** [comment]
@jayvdb  I was able to resolve it when I first got that error in my UBUNTU.

**Comment by jayvdb** [comment]
@jack17529 , I have unassigned you so that you follow the newcomers guide, which is to *finish* solving issues of difficulty newcomer and low before assigning yourself to medium difficulty issues.
